AUR eww-git packages giving errors

```
[rounak@archissexy 14:27 ~]$ yay -S eww-git
:: Checking for conflicts...
:: Checking for inner conflicts...
 -> Package conflicts found:
 -> Installing rustup will remove: rust (cargo), rust (rustfmt), rust
 -> Conflicting packages will have to be confirmed manually
[Repo Make:1]  rustup-1.25.2-1
[Aur:1]  eww-git-0.1.0.r164.g70285e0-2

:: Remove make dependencies after install? [y/N] y
  1 eww-git                                  (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> N
:: PKGBUILD up to date, Skipping (1/0): eww-git
  1 eww-git                                  (Build Files Exist)
==> Diffs to show?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> N
:: (1/1) Parsing SRCINFO: eww-git
[sudo] password for rounak: 
resolving dependencies...
looking for conflicting packages...
:: rustup and rust are in conflict. Remove rust? [y/N] y

Packages (2) rust-1:1.68.0-1 [removal]  rustup-1.25.2-1

Total Installed Size:     8.29 MiB
Net Upgrade Size:      -507.75 MiB

:: Proceed with installation? [Y/n] y
(1/1) checking keys in keyring                     [##########################] 100%
(1/1) checking package integrity                   [##########################] 100%
(1/1) loading package files                        [##########################] 100%
(1/1) checking for file conflicts                  [##########################] 100%
(2/2) checking available disk space                [##########################] 100%
:: Processing package changes...
(1/1) removing rust                                [##########################] 100%
(1/1) installing rustup                            [##########################] 100%
You may need to run rustup update stable
and possibly also rustup self upgrade-data
Optional dependencies for rustup
    lldb: rust-lldb script
    gdb: rust-gdb script [installed]
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Refreshing PackageKit...
==> Making package: eww-git 0.1.0.r164.g70285e0-2 (Saturday 18 March 2023 02:27:29 PM)
==> Retrieving sources...
  -> Updating eww git repo...
==> Validating source files with md5sums...
    eww ... Skipped
 -> rustup not satisfied, flushing install queue
==> Making package: eww-git 0.1.0.r164.g70285e0-2 (Saturday 18 March 2023 02:27:30 PM)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Updating eww git repo...
==> Validating source files with md5sums...
    eww ... Skipped
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Creating working copy of eww git repo...
Cloning into 'eww'...
done.
==> Starting pkgver()...
==> Updated version: eww-git 0.4.0.r21.g45154bb-1
==> Sources are ready.
==> Making package: eww-git 0.4.0.r21.g45154bb-1 (Saturday 18 March 2023 02:27:33 PM)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting pkgver()...
==> Removing existing $pkgdir/ directory...
==> Starting build()...
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'

  nightly-x86_64-unknown-linux-gnu unchanged - rustc 1.70.0-nightly (13afbdaa0 2023-03-17)

info: self-update is disabled for this build of rustup
info: any updates to rustup will need to be fetched with your system package manager
   Compiling proc-macro2 v1.0.51
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.5
   Compiling syn v1.0.103
   Compiling libc v0.2.136
   Compiling autocfg v1.1.0
   Compiling serde_derive v1.0.147
   Compiling serde v1.0.147
   Compiling smallvec v1.10.0
   Compiling heck v0.4.0
   Compiling pkg-config v0.3.25
   Compiling cfg-expr v0.11.0
   Compiling version-compare v0.1.0
   Compiling cfg-if v1.0.0
   Compiling once_cell v1.17.1
   Compiling thiserror v1.0.37
   Compiling version_check v0.9.4
   Compiling crunchy v0.2.2
   Compiling getrandom v0.2.8
   Compiling tiny-keccak v2.0.2
   Compiling memchr v2.5.0
   Compiling proc-macro-hack v0.5.19
   Compiling log v0.4.17
   Compiling futures-core v0.3.25
   Compiling bitflags v1.3.2
   Compiling proc-macro-error-attr v1.0.4
   Compiling slab v0.4.7
   Compiling lock_api v0.4.9
   Compiling parking_lot_core v0.9.4
   Compiling pin-project-lite v0.2.9
   Compiling futures-sink v0.3.25
   Compiling futures-task v0.3.25
   Compiling proc-macro-error v1.0.4
   Compiling futures-util v0.3.25
   Compiling pin-utils v0.1.0
   Compiling anyhow v1.0.66
   Compiling futures-channel v0.3.25
   Compiling const-random-macro v0.1.15
   Compiling indexmap v1.9.1
   Compiling siphasher v0.3.10
   Compiling memoffset v0.6.5
   Compiling scopeguard v1.1.0
   Compiling thiserror-impl v1.0.37
   Compiling futures-macro v0.3.25
   Compiling dirs-sys-next v0.1.2
   Compiling dirs-next v2.0.0
   Compiling aho-corasick v0.7.19
   Compiling futures-io v0.3.25
   Compiling futures-executor v0.3.25
   Compiling hashbrown v0.12.3
   Compiling crossbeam-utils v0.8.12
   Compiling regex-syntax v0.6.27
   Compiling parking_lot v0.12.1
   Compiling term v0.7.0
   Compiling regex v1.6.0
   Compiling toml v0.5.9
   Compiling phf_shared v0.10.0
   Compiling const-random v0.1.15
   Compiling num_cpus v1.13.1
   Compiling precomputed-hash v0.1.1
   Compiling system-deps v6.0.3
   Compiling proc-macro-crate v1.2.1
   Compiling new_debug_unreachable v1.0.4
   Compiling fixedbitset v0.4.2
   Compiling glib-sys v0.15.10
   Compiling gobject-sys v0.15.10
   Compiling gio-sys v0.15.10
   Compiling gdk-sys v0.15.1
   Compiling glib-macros v0.15.11
   Compiling pango-sys v0.15.10
   Compiling cairo-sys-rs v0.15.1
   Compiling gdk-pixbuf-sys v0.15.10
   Compiling bit-vec v0.6.3
   Compiling either v1.8.0
   Compiling ucd-trie v0.1.5
   Compiling bit-set v0.5.3
   Compiling pest v2.4.0
   Compiling itertools v0.10.5
   Compiling glib v0.15.12
   Compiling petgraph v0.6.2
   Compiling atk-sys v0.15.1
   Compiling string_cache v0.8.4
   Compiling ahash v0.3.8
   Compiling lalrpop-util v0.19.8
   Compiling ascii-canvas v3.0.0
   Compiling ena v0.14.0
   Compiling atty v0.2.14
   Compiling diff v0.1.13
   Compiling strsim v0.10.0
   Compiling ident_case v1.0.1
   Compiling fnv v1.0.7
   Compiling pico-args v0.4.2
   Compiling unicode-xid v0.2.4
   Compiling darling_core v0.14.3
   Compiling lalrpop v0.19.8
   Compiling chumsky v0.8.0
   Compiling semver-parser v0.10.2
   Compiling mio v0.8.5
   Compiling rand_core v0.6.4
   Compiling crossbeam-epoch v0.9.11
   Compiling num-traits v0.2.15
   Compiling tokio v1.21.2
   Compiling ppv-lite86 v0.2.16
   Compiling gio v0.15.12
   Compiling rustversion v1.0.9
   Compiling rand_chacha v0.3.1
   Compiling semver v0.11.0
   Compiling jaq-parse v0.9.0
   Compiling darling_macro v0.14.3
   Compiling bincode v1.3.3
   Compiling tokio-macros v1.8.0
   Compiling ahash v0.7.6
   Compiling socket2 v0.4.7
   Compiling signal-hook-registry v1.4.0
   Compiling num-integer v0.1.45
   Compiling termcolor v1.1.3
   Compiling ref-cast v1.0.12
   Compiling serde_json v1.0.87
   Compiling rayon-core v1.9.3
   Compiling async-trait v0.1.64
   Compiling io-lifetimes v1.0.2
   Compiling bytes v1.2.1
   Compiling jaq-std v0.9.0
   Compiling darling v0.14.3
   Compiling rustc_version v0.3.3
   Compiling rand v0.8.5
   Compiling crossbeam-channel v0.5.6
   Compiling gtk-sys v0.15.3
   Compiling ref-cast-impl v1.0.12
   Compiling phf_shared v0.9.0
   Compiling x11 v2.20.0
   Compiling num-bigint v0.4.3
   Compiling convert_case v0.4.0
   Compiling lazy_static v1.4.0
   Compiling ryu v1.0.11
   Compiling rustix v0.36.3
   Compiling cached_proc_macro_types v0.1.0
   Compiling itoa v1.0.4
   Compiling cached_proc_macro v0.16.0
   Compiling gdk-pixbuf v0.15.11
   Compiling derive_more v0.99.17
   Compiling phf_generator v0.9.1
   Compiling strum_macros v0.24.3
   Compiling crossbeam-deque v0.8.2
   Compiling field-offset v0.3.4
   Compiling simplexpr v0.1.0 (/home/rounak/.cache/yay/eww-git/src/eww/crates/simplexpr)
   Compiling pango v0.15.10
   Compiling cairo-rs v0.15.12
   Compiling gdkx11-sys v0.15.1
   Compiling futures v0.3.25
   Compiling instant v0.1.12
   Compiling rayon v1.5.3
   Compiling num-rational v0.4.1
   Compiling linux-raw-sys v0.1.3
   Compiling dyn-clone v1.0.10
   Compiling hashbrown v0.13.2
   Compiling async_once v0.2.6
   Compiling quick-error v1.2.3
   Compiling humantime v1.3.0
   Compiling cached v0.42.0
   Compiling jaq-core v0.9.0
   Compiling gdk v0.15.4
   Compiling strum v0.24.1
   Compiling eww_shared_util v0.1.0 (/home/rounak/.cache/yay/eww-git/src/eww/crates/eww_shared_util)
   Compiling phf_macros v0.9.0
   Compiling hashbrown v0.11.2
   Compiling yuck v0.1.0 (/home/rounak/.cache/yay/eww-git/src/eww/crates/yuck)
   Compiling nix v0.24.2
   Compiling inotify-sys v0.1.5
   Compiling gtk v0.15.5
   Compiling sysinfo v0.26.5
   Compiling unicode-width v0.1.10
   Compiling static_assertions v1.1.0
   Compiling os_str_bytes v6.3.0
   Compiling same-file v1.0.6
   Compiling clap_lex v0.3.0
   Compiling walkdir v2.3.2
error[E0557]: feature has been removed
 --> crates/simplexpr/src/lib.rs:3:12
  |
3 | #![feature(box_syntax)]
  |            ^^^^^^^^^^ feature has been removed
  |
  = note: replaced with `#[rustc_box]`

error: `box_syntax` has been removed
  --> crates/simplexpr/src/eval.rs:96:58
   |
96 | ... box b) => BinOp(span, box a.try_map_var_refs(f)?, op, box b.try_map_var_...
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use `Box::new()` instead
   |
96 |             BinOp(span, box a, op, box b) => BinOp(span, Box::new(a.try_map_var_refs(f)?), op, box b.try_map_var_refs(f)?),
   |                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: `box_syntax` has been removed
  --> crates/simplexpr/src/eval.rs:96:90
   |
96 | ..._map_var_refs(f)?, op, box b.try_map_var_refs(f)?),
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use `Box::new()` instead
   |
96 |             BinOp(span, box a, op, box b) => BinOp(span, box a.try_map_var_refs(f)?, op, Box::new(b.try_map_var_refs(f)?)),
   |                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: `box_syntax` has been removed
  --> crates/simplexpr/src/eval.rs:98:59
   |
98 | ...) => UnaryOp(span, op, box a.try_map_var_refs(f)?),
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use `Box::new()` instead
   |
98 |             UnaryOp(span, op, box a) => UnaryOp(span, op, Box::new(a.try_map_var_refs(f)?)),
   |                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: `box_syntax` has been removed
   --> crates/simplexpr/src/eval.rs:100:30
    |
100 | ...   IfElse(span, box a.try_map_var_refs(f)?, box b.try_map_var_refs(f)?, ...
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: use `Box::new()` instead
    |
100 |                 IfElse(span, Box::new(a.try_map_var_refs(f)?), box b.try_map_var_refs(f)?, box c.try_map_var_refs(f)?)
    |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: `box_syntax` has been removed
   --> crates/simplexpr/src/eval.rs:100:58
    |
100 | ....try_map_var_refs(f)?, box b.try_map_var_refs(f)?, box c.try_map_var_ref...
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: use `Box::new()` instead
    |
100 |                 IfElse(span, box a.try_map_var_refs(f)?, Box::new(b.try_map_var_refs(f)?), box c.try_map_var_refs(f)?)
    |                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: `box_syntax` has been removed
   --> crates/simplexpr/src/eval.rs:100:86
    |
100 | ....try_map_var_refs(f)?, box c.try_map_var_refs(f)?)
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: use `Box::new()` instead
    |
100 |                 IfElse(span, box a.try_map_var_refs(f)?, box b.try_map_var_refs(f)?, Box::new(c.try_map_var_refs(f)?))
    |                                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: `box_syntax` has been removed
   --> crates/simplexpr/src/eval.rs:103:40
    |
103 | ...   JsonAccess(span, safe, box a.try_map_var_refs(f)?, box b.try_map_var_...
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: use `Box::new()` instead
    |
103 |                 JsonAccess(span, safe, Box::new(a.try_map_var_refs(f)?), box b.try_map_var_refs(f)?)
    |                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: `box_syntax` has been removed
   --> crates/simplexpr/src/eval.rs:103:68
    |
103 | ....try_map_var_refs(f)?, box b.try_map_var_refs(f)?)
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: use `Box::new()` instead
    |
103 |                 JsonAccess(span, safe, box a.try_map_var_refs(f)?, Box::new(b.try_map_var_refs(f)?))
    |                                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

   Compiling codespan-reporting v0.11.1
For more information about this error, try `rustc --explain E0557`.
error: could not compile `simplexpr` (lib) due to 9 previous errors
warning: build failed, waiting for other jobs to finish...
==> ERROR: A failure occurred in build().
    Aborting...
checking dependencies...

Packages (1) rustup-1.25.2-1

Total Removed Size:  8.29 MiB

:: Do you want to remove these packages? [Y/n] 
:: Processing package changes...
(1/1) removing rustup                              [##########################] 100%
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Refreshing PackageKit...
 -> error making: eww-git
[rounak@archissexy 14:28 ~]$ rustup --explain E0557
bash: rustup: command not found
[rounak@archissexy 14:28 ~]$ rustc --explain E0557
bash: /usr/bin/rustc: No such file or directory
```

--- a/crates/eww/src/main.rs
+++ b/crates/eww/src/main.rs
@@ -1,6 +1,5 @@
 #![feature(trace_macros)]
 #![feature(drain_filter)]
-#![feature(box_syntax)]
 #![feature(box_patterns)]
 #![feature(slice_concat_trait)]
 #![feature(try_blocks)]

--- a/crates/simplexpr/src/eval.rs
+++ b/crates/simplexpr/src/eval.rs
@@ -93,14 +93,14 @@ impl SimplExpr {
     pub fn try_map_var_refs<E, F: Fn(Span, VarName) -> Result<SimplExpr, E> + Copy>(self, f: F) -> Result<Self, E> {
         use SimplExpr::*;
         Ok(match self {
-            BinOp(span, box a, op, box b) => BinOp(span, box a.try_map_var_refs(f)?, op, box b.try_map_var_refs(f)?),
+            BinOp(span, box a, op, box b) => BinOp(span, Box::new(a.try_map_var_refs(f)?), op, Box::new(b.try_map_var_refs(f)?)),
             Concat(span, elems) => Concat(span, elems.into_iter().map(|x| x.try_map_var_refs(f)).collect::<Result<_, _>>()?),
-            UnaryOp(span, op, box a) => UnaryOp(span, op, box a.try_map_var_refs(f)?),
+            UnaryOp(span, op, box a) => UnaryOp(span, op, Box::new(a.try_map_var_refs(f)?)),
             IfElse(span, box a, box b, box c) => {
-                IfElse(span, box a.try_map_var_refs(f)?, box b.try_map_var_refs(f)?, box c.try_map_var_refs(f)?)
+                IfElse(span, Box::new(a.try_map_var_refs(f)?), Box::new(b.try_map_var_refs(f)?), Box::new(c.try_map_var_refs(f)?))
             }
             JsonAccess(span, safe, box a, box b) => {
-                JsonAccess(span, safe, box a.try_map_var_refs(f)?, box b.try_map_var_refs(f)?)
+                JsonAccess(span, safe, Box::new(a.try_map_var_refs(f)?), Box::new(b.try_map_var_refs(f)?))
             }
             FunctionCall(span, name, args) => {
                 FunctionCall(span, name, args.into_iter().map(|x| x.try_map_var_refs(f)).collect::<Result<_, _>>()?)

--- a/crates/simplexpr/src/lib.rs
+++ b/crates/simplexpr/src/lib.rs
@@ -1,6 +1,5 @@
 #![feature(box_patterns)]
 #![feature(pattern)]
-#![feature(box_syntax)]
 #![feature(try_blocks)]
 #![feature(unwrap_infallible)]
 #![feature(never_type)]

